File: contest/joined/codeforces-1846/E2.cpp
// Author : ysh
// 2023/07/07 Fri 23:27:53
#include<bits/stdc++.h>
using namespace std;
#define int long long
signed main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int n;cin>>n;
    while(n--) {
        long long n;cin>>n;
        bool o = 0;
        for(int i = 2;i<=64;i++) {
            if(o) continue;
            long long tmp = pow(n,1.0 / i);
            for(long long j = max(2LL,tmp - 3);j <= tmp + 3;j++) {
                int ans = 0;
                int last = 1;
                for(int k = 0;k<=i;k++) ans = ans + last,last = last * j;
                if(ans == n) o = 1;
            }
        }
        if(o) cout<<"YES\n";
        else cout<<"NO\n";
    }
    return 0;
}

File: contest/joined/TOPC-2023-PRE/E2.py
# Author : ysh
# 2023/09/16 Sat 11:51:56
a,b = [int(i) for i in input().split()]

i,j,k = a,a * a - 4,2

def check(a,j,k):
    if j == 1:
        return a
    tmp = check(a,j >> 1,k)